src/components/data-management/DownloadData.jsx

```diff
@@ -0,0 +1,171 @@
+import downloadTypes from 'utils/data-management/downloadTypes';
+import { getFromApiExpectOK } from 'utils/getDataExpectOK';
+import pushNotificationMessage from 'utils/pushNotificationMessage';
+import endUserMessages from 'utils/endUserMessages';
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import { exportQCParameters, filterQCParameters } from '../../utils/data-management/exportQCParameters';
+import { loadBackendStatus } from '../../redux/actions/backendStatus/index';
+
+const DownloadData = (props) => {
+  const {
+    activeProjectUuid,
+  } = props;
+  const dispatch = useDispatch();
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
+  const experimentSettings = useSelector((state) => state.experimentSettings);
+  const backendStatus = useSelector((state) => state.backendStatus);
+  const samples = useSelector((state) => state.samples);
+  const projects = useSelector((state) => state.projects);
+
+  // Change if we have more than one experiment per project
+  const experimentId = activeProject?.experiments[0];
+
+  useEffect(() => {
+    if (experimentId && !backendStatus[experimentId]) {
+      dispatch(loadBackendStatus(experimentId));
+    }
+  }, [experimentId]);
+
+  const getQcHasRun = () => (
+    experimentId
+    && (backendStatus[experimentId]?.status.pipeline?.status === pipelineStatus.SUCCEEDED)
+  );
+
+  const getGem2sHasRun = () => (
+    experimentId
+    && (backendStatus[experimentId]?.status?.gem2s?.status === pipelineStatus.SUCCEEDED)
+  );
+
+  const getAllSamplesAnalysed = () => {
+    // Returns true only if there is at least one sample in the currently active
+    // project AND all samples in the project have been analysed.
+    if (!activeProject?.samples?.length) {
+      return false;
+    }
+    const steps = Object.values(_.omit(experimentSettings?.processing, ['meta']));
+
+    return steps.length > 0
+      // eslint-disable-next-line no-prototype-builtins
+      && activeProject?.samples?.every((s) => steps[0].hasOwnProperty(s));
+  };
+
+  const gem2sHasRun = getGem2sHasRun();
+  const qcHasRun = getQcHasRun();
+  const allSamplesAnalysed = getAllSamplesAnalysed();
+
+  const downloadExperimentData = async (type) => {
+    try {
+      if (!experimentId) throw new Error('No experimentId specified');
+      if (!Object.values(downloadTypes).includes(type)) throw new Error('Invalid download type');
+
+      const { signedUrl } = await getFromApiExpectOK(`/v1/experiments/${experimentId}/download/${type}`);
+      const link = document.createElement('a');
+      link.style.display = 'none';
+      link.href = signedUrl;
+
+      document.body.appendChild(link);
+      link.click();
+
+      setTimeout(() => {
+        URL.revokeObjectURL(link.href);
+        link.parentNode.removeChild(link);
+      }, 0);
+    } catch (e) {
+      pushNotificationMessage('error', endUserMessages.ERROR_DOWNLOADING_DATA);
+    }
+  };
+
+  return (
```

Comment:
There is a bug in this code when I tested in staging: if I click the `DownloadData` button, the website crashes, showing this error:
<img width="584" alt="Screenshot 2021-08-30 at 10 59 42" src="https://user-images.githubusercontent.com/65414651/131314665-17269fe4-07e7-4eb6-935e-b97bef3e23f8.png">

Reply:
When you fix it, make sure you also include a unit test to test for this bug and prevent us from having it again.